### PR TITLE
Adds new Special Syndicate Bundle, "Ancient"

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request

The Syndicate loves a good throwback, and this <s>new</s> old kit is about as far back as the Syndicate goes.

The Ancient bundle contains all the uplink items circa 2006, plus an extra Telecrystal to purchase a Failsafe code or a C4 to self-destruct your uplink with. Just like old times.

## Why It's Good For The Game

_sips Monkey Energy_
Mhm.

## Changelog
:cl: YakumoChen
add: A less-than-new Syndicate Special bundle reminds you of the good old days when we didn't need all those newfangled traitor items the young-uns get. We had 6 items in the uplink and we had Monkey in rotation and by god we made do.
/:cl:

### CODE REVIEW

**Comment by YakumoChen** [comment]
Disclaimer: One of the items from 2006 is a "cloaking device" which is hard to tell the exact function of from the code, so I may have taken liberties with which item to represent it with.

**Comment by necromanceranne** [comment]
I'd imagine it's the infamous cloaker. It's still present on Goon, and basically turns you invisible. As far as I know on the goon version, only silicons and the detective's thermals (when he had thermals) could see through the cloak.

We do have an item _similar_ to it. The shadowcloak belt.

**Comment by YakumoChen** [comment]
Sounds like the projector is the best fitting representation I can fill it with, since it basically also makes you invisible and has the same apparent weaknesses as the "cloaking device" does in this OpenSS13 code.

File: code/game/objects/items/storage/uplink_kits.dm [hunk 1]
```diff
@@ -195,6 +195,15 @@
...
 			new /obj/item/melee/transforming/energy/sword/saber/blue(src) //see see it fits the theme bc its blue and ice is blue
+			
+		if("ancient") //A kit so old, it's probably older than you.
+			new /obj/item/card/emag(src)
+			new /obj/item/pen/sleepy(src)
+			new /obj/item/reagent_containers/pill/cyanide
+			new /obj/item/chameleon(src)
+			new /obj/item/gun/ballistic/revolver(src)
+			new /obj/item/implanter/freedom(src)
+			new /obj/item/stack/telecrystal(src) //The failsafe/self destruct isn't an item we can physically include in the kit, but 1 TC is technically enough to buy the equivalent.
```
Comment (ATH1909) [inline code/game/objects/items/storage/uplink_kits.dm:206]: maybe this kit could come with a piece of paper that explains what this telecrystal is intended to be used for?

File: code/game/objects/items/storage/uplink_kits.dm [hunk 2]
```diff
@@ -195,6 +195,15 @@
 			new /obj/item/dnainjector/cryokinesis(src)
 			new /obj/item/gun/energy/temperature/security(src)
 			new /obj/item/melee/transforming/energy/sword/saber/blue(src) //see see it fits the theme bc its blue and ice is blue
+			
+		if("ancient") //A kit so old, it's probably older than you.
+			new /obj/item/card/emag(src)
+			new /obj/item/pen/sleepy(src)
+			new /obj/item/reagent_containers/pill/cyanide
+			new /obj/item/chameleon(src)
```
Comment (ATH1909) [inline code/game/objects/items/storage/uplink_kits.dm:203]: you might want to mention in a comment here that this is a substitute for the cloaker

File: code/game/objects/items/storage/uplink_kits.dm [hunk 3]
```diff
@@ -195,6 +195,15 @@
 			new /obj/item/dnainjector/cryokinesis(src)
 			new /obj/item/gun/energy/temperature/security(src)
 			new /obj/item/melee/transforming/energy/sword/saber/blue(src) //see see it fits the theme bc its blue and ice is blue
+			
+		if("ancient") //A kit so old, it's probably older than you.
+			new /obj/item/card/emag(src)
+			new /obj/item/pen/sleepy(src)
+			new /obj/item/reagent_containers/pill/cyanide
```
Comment (coiax) [inline code/game/objects/items/storage/uplink_kits.dm:202]:
```suggestion
			new /obj/item/reagent_containers/pill/cyanide(src)
```

File: code/game/objects/items/storage/uplink_kits.dm (hunk 1 above)
Comment (coiax) [reply code/game/objects/items/storage/uplink_kits.dm:206]: I am fairly sure that it's possible to have the self destruct included in the kit on purchase, but it'll be a little fiddly.

File: code/game/objects/items/storage/uplink_kits.dm [hunk 4]
```diff
@@ -120,7 +120,7 @@
 			new /obj/item/card/emag/doorjack(src) // 3 tc
 
 /obj/item/storage/box/syndicate/bundle_b/PopulateContents()
-	switch (pickweight(list( "bond" = 2, "ninja" = 1, "darklord" = 1, "white_whale_holy_grail" = 2, "mad_scientist" = 2, "bee" = 1, "mr_freeze" = 2)))
+	switch (pickweight(list( "bond" = 2, "ninja" = 1, "darklord" = 1, "white_whale_holy_grail" = 2, "mad_scientist" = 2, "bee" = 1, "mr_freeze" = 2, "ancient" = 1)))
```
Comment (coiax) [inline code/game/objects/items/storage/uplink_kits.dm:123]: Split the selection out into its own line, make the weights across a multi-line list, and make these local file defines.

Comment (YakumoChen) [reply code/game/objects/items/storage/uplink_kits.dm:123]: > defines

That's going to take me at least a couple hours to figure out how to write defines, at minimum...

File: code/game/objects/items/storage/uplink_kits.dm (hunk 1 above)
Comment (YakumoChen) [reply code/game/objects/items/storage/uplink_kits.dm:206]: Yeah, but ik personally not cool enough to know how to do this.

**Comment by ArcaneMusic** [comment]
You might want to squash your commits here, since you've got 74 commits on a 10 line change.

**Comment by YakumoChen** [comment]
I had some kind of weird conflict trying to update my branch
<s>I dont actually know how to squash commits</s>

**Comment by Krysonism** [comment]
Maybe include the shadow cloak instead of the chameleon projector?

**Comment by YakumoChen** [comment]
I would rather trust necromanceranne that it is "horrendously overpowered" and be happy with the projector as a substitute.

**Comment by cacogen** [comment]
SOUL

**Comment by YakumoChen** [comment]
> SOUL

This just game me the idea to provide the kit in an old red toolbox instead for style points.

**Comment by YakumoChen** [comment]
ok, the failed checks are no longer relevant to my PR.

File: code/game/objects/items/storage/uplink_kits.dm [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+#define KIT_LORD_SINGULOTH "lordsingulo"
+
+#define KIT_JAMES_BOND "bond"
+#define KIT_NINJA "ninja"
+#define KIT_DARK_LORD "darklord"
+#define KIT_WHITE_WHALE_HOLY_GRAIL "white_whale_holy_grail"
+#define KIT_MAD_SCIENTIST "mad_scientist"
+#define KIT_BEES "bee"
+#define KIT_MR_FREEZE "mr_freeze"
+#define KIT_TRAITOR_2006 "ancient"
```
Comment (Mothblocks) [inline code/game/objects/items/storage/uplink_kits.dm:21]: Undef all these at the end.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/uplink_kits.dm (hunk 5 above)
Comment (YakumoChen) [reply code/game/objects/items/storage/uplink_kits.dm:21]: what?

**Pull request merged**